hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -189,38 +170,73 @@ def is_cancel_request_in_exchange_synchronous(self) -> bool:
     def is_trading_required(self) -> bool:
         return self._trading_required
 
+    @property
+    def trading_pairs_request_path(self) -> str:
+        return ""
+
+    @property
+    def check_network_request_path(self) -> str:
+        return ""
+
+    def _api_request(
+        self,
+        path_url,
+        overwrite_url: Optional[str] = None,
+        method: RESTMethod = RESTMethod.GET,
+        params: Optional[Dict[str, Any]] = None,
+        data: Optional[Dict[str, Any]] = None,
+        is_auth_required: bool = False,
+        return_err: bool = False,
+        limit_id: Optional[str] = None,
+        **kwargs,
+    ) -> Dict[str, Any]:
+        raise NotImplementedError
+
+    def authenticator(self):
+        return AuthBase()
+
+    async def check_network(self) -> NetworkStatus:
+        await self._testnet_provider.wait_connect()
```

Comment:
should probably not wait on the connection here? If the connection's not ready, then I expect this should return `NOT_CONNECTED`, not wait until the connection is ready always

Reply:
I think here it doesn't make a difference because there is a timeout error on the side of `_check_network_loop` in `network_iterator.pyx`. I guess it would check if the provider is ready more times.

But to me, the fact that there is a timeout error in the background means that we should take the time to check what we need to, and the program will handle it if it is taking too long.

Reply:
Probably depends on the event loop and how it works here. I expect most likely it'd be like what you said; that the network loop would just check more times if it got false back. Also something that I think could be fine to wait for the hummingbot team to let us know what should be done.